tests/unit/amo/components/TestCollectionManager.js

```diff
@@ -304,6 +304,38 @@ describe(__filename, () => {
       .toHaveProp('disabled', true);
   });
 
+  it('trims leading and trailing spaces from slug and name before submitting', () => {
+    const authorUsername = 'collection-owner';
+    const name = 'trishul';
+    const slug = 'trishul';
+    const lang = 'en-US';
+    dispatchSignInActions({ lang, store });
+
+    const collection = createInternalCollection({
+      detail: createFakeCollectionDetail({ authorUsername, name, slug }),
+    });
+
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const root = render({ collection });
+
+    // Enter in collection name and slug with trailing and leading spaces.
+    typeInput({ root, name: 'name', text: '  trishul   ' });
+    typeInput({ root, name: 'slug', text: '  trishul   ' });
+
+    dispatchSpy.reset();
+    simulateSubmit(root);
+
+    sinon.assert.calledWith(dispatchSpy, updateCollection({
+      collectionSlug: collection.slug,
```

Comment:
`collection.slug` could just be `slug`.

Reply:
I am already creating `createFakeCollectionDetail` with custom name and slug that's why I used `collection.name` and `collection.slug`

Reply:
Right, which is why I suggested you could shorten it to just `slug`. If `collection.slug` is always going to be the same as `slug`, why use the longer `collection.slug`?

Reply:
done :)